File: scheduler/serve.py
"""
serve.py — Servidor local para el sitio del Mundial 2026 (Python)
Puerto: 3026
Uso: python scheduler/serve.py
"""

import http.server
import socketserver
import os
import sys

PORT = 3026
ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

class Handler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=ROOT, **kwargs)

    def log_message(self, format, *args):
        pass  # silenciar logs

os.chdir(ROOT)

with socketserver.TCPServer(("", PORT), Handler) as httpd:
    print(f"\n⚽ Mundial 2026 — Servidor iniciado")
    print(f"🌐 http://localhost:{PORT}")
    print(f"📁 Sirviendo: {ROOT}")
    print(f"\nPresiona Ctrl+C para detener.\n")
    sys.stdout.flush()
    try:
        httpd.serve_forever()
    except KeyboardInterrupt:
        print("\nServidor detenido.")

File: scheduler/serve.js
/**
 * serve.js — Servidor local para el sitio del Mundial 2026
 * Puerto: 3026
 * Uso: node scheduler/serve.js
 */

const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = 3026;
const ROOT = path.join(__dirname, '..');

const MIME = {
  '.html': 'text/html; charset=utf-8',
  '.css':  'text/css; charset=utf-8',
  '.js':   'application/javascript; charset=utf-8',
  '.json': 'application/json; charset=utf-8',
  '.png':  'image/png',
  '.jpg':  'image/jpeg',
  '.svg':  'image/svg+xml',
  '.ico':  'image/x-icon',
  '.webp': 'image/webp',
};

const server = http.createServer((req, res) => {
  let urlPath = req.url.split('?')[0];
  if (urlPath === '/') urlPath = '/index.html';

  const filePath = path.join(ROOT, urlPath);

  // Seguridad: no salir del directorio raíz
  if (!filePath.startsWith(ROOT)) {
    res.writeHead(403);
    return res.end('Forbidden');
  }

  fs.readFile(filePath, (err, data) => {
    if (err) {
      res.writeHead(404, { 'Content-Type': 'text/plain' });
      return res.end('Not found: ' + urlPath);
    }

    const ext = path.extname(filePath);
    const contentType = MIME[ext] || 'application/octet-stream';

    res.writeHead(200, {
      'Content-Type': contentType,
      'Cache-Control': 'no-cache, no-store',
      'Access-Control-Allow-Origin': '*',
    });
    res.end(data);
  });
});

server.listen(PORT, () => {
  console.log(`\n⚽ Mundial 2026 — Servidor iniciado`);
  console.log(`🌐 http://localhost:${PORT}`);
  console.log(`📁 Sirviendo: ${ROOT}`);
  console.log(`\nPresiona Ctrl+C para detener.\n`);
});

server.on('error', err => {
  if (err.code === 'EADDRINUSE') {
    console.error(`Puerto ${PORT} en uso. Intenta: http://localhost:${PORT}`);
  } else {
    console.error('Error del servidor:', err.message);
  }
  process.exit(1);
});
